# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2005/02/02 17:41:06-08:00 guninski@guninski.com 
#   [PATCH] Fix sign checks in copy_from_read_buf()
#   
#   Fix signedness and remove the now unnecessary cast.
#   
#   Acked-by: Marcelo Tosatti <marcelo.tosatti@cyclades.com>
#   Signed-off-by: Linus Torvalds <torvalds@osdl.org>
# 
# drivers/char/n_tty.c
#   2005/01/30 07:56:05-08:00 guninski@guninski.com +2 -2
#   Fix sign checks in copy_from_read_buf()
# 
diff -Nru a/drivers/char/n_tty.c b/drivers/char/n_tty.c
--- a/drivers/char/n_tty.c	2005-02-18 04:40:54 -08:00
+++ b/drivers/char/n_tty.c	2005-02-18 04:40:54 -08:00
@@ -1143,13 +1143,13 @@
 
 {
 	int retval;
-	ssize_t n;
+	size_t n;
 	unsigned long flags;
 
 	retval = 0;
 	spin_lock_irqsave(&tty->read_lock, flags);
 	n = min(tty->read_cnt, N_TTY_BUF_SIZE - tty->read_tail);
-	n = min((ssize_t)*nr, n);
+	n = min(*nr, n);
 	spin_unlock_irqrestore(&tty->read_lock, flags);
 	if (n) {
 		mb();
